### PR TITLE
Update config template and validations

### PR DESCRIPTION
**Pull request by CunliangGeng**
change config bigscape.cutoff from float to string

transform config `root_dir` to full path.

### CODE REVIEW

**Comment by CunliangGeng** [comment]
* **#218** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/218?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#217** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/217?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#215** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/215?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#214** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/214?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#212** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/212?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#211** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/211?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#210** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/210?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#209** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/209?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#208** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/208?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `dev`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @CunliangGeng and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: src/nplinker/config.py [hunk 1]
```diff
@@ -10,6 +11,7 @@
 # directory.
 # The loaded config data is available by importing this module and accessing the 'config' variable.
 
+__all__ = ["config"]
```
Comment (gcroci2) [inline src/nplinker/config.py:14]: Why is this needed?

**Review by gcroci2 — APPROVED** [review]

File: src/nplinker/config.py (hunk 1 above)
Comment (CunliangGeng) [reply src/nplinker/config.py:14]: To make sure only the variable `config` is imported when using `from nplinker.config import *`. We don't want the other variables like `validators` imported.

**Comment by CunliangGeng** [comment]
### Merge activity

* **Mar 1, 9:49 AM EST**: @CunliangGeng started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/NPLinker/nplinker/209).
* **Mar 1, 9:50 AM EST**: [Graphite](https://app.graphite.dev/github/pr/NPLinker/nplinker/209) rebased this pull request as part of a merge.
* **Mar 1, 9:51 AM EST**: @CunliangGeng merged this pull request with [Graphite](https://app.graphite.dev/github/pr/NPLinker/nplinker/209).
